typescript/src/schema/api.ts

```diff
@@ -74,7 +74,8 @@ export class API {
           options.grpcServiceConfig,
           allResourceDatabase,
           resourceDatabase,
-          options.bundleConfigs
+          options.bundleConfigs,
+          options.iamService
```

Comment:
This is going out of control here. Let's stop this now :)  Can I ask you to export an `interface ProtoParameters` from `proto.ts` and make the `Proto` constructor accept one object?

```ts
map[fd.name!] = new Proto({
  fd,
  packageName,
  grpcServiceConfig: options.grpcServiceConfig,
  ...
});
```

Reply:
sure! we can reuse the `naming.Options` interface. Thanks! same below.